### PR TITLE
Improving azure storage

### PR DESCRIPTION
**Pull request by erlingbo**

### CODE REVIEW

File: storages/backends/azure_storage.py [hunk 1]
```diff
@@ -57,9 +57,20 @@ def size(self, name):
         return properties["content-length"]
 
     def _save(self, name, content):
+        if hasattr(content.file, 'content_type'):
```
Comment (jschneier) [inline storages/backends/azure_storage.py]: This assumes we have a `ContentFile` right? I know that's what we return from `_open` but is this going to be the case generally?

**Comment by jschneier** [comment]
Thanks for your contribution! I've left one concern of mine in line but looks nice other than that.

File: storages/backends/azure_storage.py (hunk 1 above)
Comment (erlingbo) [inline storages/backends/azure_storage.py]: As i have interpreted the Django docs[1] a custom storage should handle File objects [2]. Thus it should always have a file attribute. This code checks whether the file has a content_type, if not it tries to guess it.

Please correct me if I have not interpreted the docs correctly ;)

[1] https://docs.djangoproject.com/en/1.8/howto/custom-file-storage/
[2] https://docs.djangoproject.com/en/1.8/ref/files/file/

Comment (jschneier) [inline storages/backends/azure_storage.py]: You are correct, sorry about that.

File: storages/backends/azure_storage.py [hunk 2]
```diff
@@ -57,9 +57,20 @@ def size(self, name):
...
+            content_type = content.file.content_type
+        else:
+            content_type = mimetypes.guess_type(name)[0]
```
Comment (tbartelmess) [inline storages/backends/azure_storage.py]: I think you forgot to `import mimetypes`

Comment (erlingbo) [inline storages/backends/azure_storage.py]: Nice spot. I'll fix :)

**Comment by pwwebdev** [comment]
Can someone merge this pull request?

**Comment by Hanan-Natan** [comment]
Merge please.

**Comment by cveilleux** [comment]
+1

**Comment by pwwebdev** [comment]
Anyone with write access who can merge this?

**Comment by jschneier** [comment]
Sorry for the delay folks, I have merged this. I realize that Azure storage probably wasn't really usable without that `url` fix in particular.

**Pull request merged**